lib/virt_autotest/utils.pm

```diff
@@ -719,4 +723,53 @@ sub is_sev_es_guest {
     }
 }
 
+# remove a vm listed via 'virsh list'
+sub remove_vm {
+    my $vm = shift;
+    my $is_persistent_vm = script_output "virsh dominfo $vm | sed -n '/Persistent:/p' | awk '{print \$2}'";
+    my $vm_state = script_output "virsh domstate $vm";
+    if ($vm_state ne "shut off") {
+        assert_script_run("virsh destroy $vm", 30);
+    }
+    if ($is_persistent_vm eq "yes") {
+        assert_script_run("virsh undefine $vm || virsh undefine $vm --keep-nvram", 30);
+    }
+}
+
+#Start the guest from the downloaded vm xml and vm disk file
+sub restore_downloaded_guests {
+    my ($guest, $vm_xml_dir) = @_;
+    record_info("Guest restored", "$guest");
+    my $vm_xml = "$vm_xml_dir/$guest.xml";
+    assert_script_run("virsh define $vm_xml", 30);
+}
+
+sub save_original_guest_xmls {
+    my ($save_dir, @guests) = @_;
+    $save_dir //= "/tmp/download_vm_xml";
+    @guests = keys %virt_autotest::common::guests if @guests == 0;
+    assert_script_run "mkdir -p $save_dir" unless script_run("ls $save_dir") == 0;
+    foreach my $guest (@guests) {
+        unless (script_run("ls $save_dir/$guest.xml") == 0) {
+            assert_script_run "virsh dumpxml --inactive $guest > $save_dir/$guest.xml";
+        }
+    }
```

Comment:
I think you can just
```
    foreach my $guest (@guests) {
            assert_script_run "virsh dumpxml --inactive $guest > $save_dir/$guest.xml";
    }
```
?

Reply:
No, the existing guest xml is prefered. the guest may not be seen in virsh(not running or even defined), especially 'assert' is used here.